src/libraries/System.Linq/src/System/Linq/ElementAt.cs

```diff
@@ -51,6 +51,87 @@ public static TSource ElementAt<TSource>(this IEnumerable<TSource> source, int i
             return default;
         }
 
+        /// <summary>Returns the element at a specified index in a sequence.</summary>
+        /// <param name="source">An <see cref="IEnumerable{T}" /> to return an element from.</param>
+        /// <param name="index">The index of the element to retrieve, which is either from the start or the end.</param>
+        /// <typeparam name="TSource">The type of the elements of <paramref name="source" />.</typeparam>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="source" /> is <see langword="null" />.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="index" /> is outside the bounds of the <paramref name="source" /> sequence.
+        /// </exception>
+        /// <returns>The element at the specified position in the <paramref name="source" /> sequence.</returns>
+        public static TSource ElementAt<TSource>(this IEnumerable<TSource> source, Index index)
+        {
+            if (source == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.source);
+            }
+
+            if (!index.IsFromEnd)
+            {
+                return source.ElementAt(index.Value);
+            }
+
+            int indexFromEnd = index.Value;
+            Debug.Assert(indexFromEnd >= 0);
+            if (indexFromEnd == 0)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.index);
```

Comment:
This throw helper call appears multiple times... it seems like the method could be restructured slightly such that we'd only need the one call at the end.

Reply:
Updated with `TryGetElementAt`.